### PR TITLE
Introduce `ruff` for code linting

### LINKED ISSUE #1273: Investigate/Integrate ruff
https://github.com/astral-sh/ruff

Seems to be fast and a replacement for flake and isort
(but not for black?)

### PR DESCRIPTION
**Pull request by chrysle**
Closes #1273.

cc  @encbladexp

### CODE REVIEW

**Review by chris34 — COMMENTED** [review]
what i see as improvements (aside from the inline comments):

 - i assume `ruff format --check` was not tested, as `'` is still used instead of `"`? (`ruff format` is ruff's equivalent of black. Black by default prefers `"` for all strings. `ruff check --fix` seems te be the replacement for tools like pyflake or isort)
   - i'm also sceptical to format everything in black-style in one PR (that could get huge). A more staged approach would be fine IMHO (PR touches a file → reformat the changed files).
   -  or we could also set `'` as default for inyoka via `quote-style = "single"`.
 - is the `setup.cfg` still needed or can it be deleted?
 - run `ruff check` (and `ruff format --check`) in github actions

Nevertheless a good start. :+1: 
Do you want to fix the points or should I?

File: inyoka/ikhaya/forms.py [hunk 1]
```diff
@@ -76,7 +82,7 @@ def save(self):
         if self.cleaned_data.get('updated', None):
             instance.updated = self.cleaned_data['updated']
         elif {'pub_date', 'pub_time'} in set(self.cleaned_data.keys()):
-            instance.updated = date_time_to_datetime(
+            instance.updated = datetime.combine(
```
[the remark this answers is not in the document]
Comment (chrysle) [reply inyoka/ikhaya/forms.py:85]: > (I suppose it was a left over from #848 and you changed it manually.)

Indeed.

[the remark this answers is not in the document]
Comment (chrysle) [reply inyoka/ikhaya/forms.py:85]: This indicates missing test coverage..

[the remark this answers is not in the document]
Comment (chris34) [reply inyoka/ikhaya/forms.py:85]: > This indicates missing test coverage..

Indeed. We can only get better. :smile: 

(for those that are curios: Every CI run generates a coverage report → f.e. https://github.com/inyokaproject/inyoka/actions/runs/10656501066 Of course this line is not covered yet)

**Pull request merged**